packages/code-studio/src/main/AppInit.tsx

```diff
@@ -184,6 +184,13 @@ function AppInit(props: AppInitProps) {
         setError(`Server shutdown: ${detail ?? 'Unknown reason'}`);
         setDisconnectError(null);
       });
+      connection.addEventListener(
+        dh.CoreClient.EVENT_RECONNECT_AUTH_FAILED,
+        event => {
+          log.warn('Reconnect authentication failed');
+          setError('Reconnect authentication failed');
```

Comment:
@dsmmcken thoughts on a message for this? The scenario is the user was disconnected for longer than the auth token is valid, so when it reconnects the authentication fails. User should be prompted to reload since this just shows the window w/ a warning triangle and message.

Maybe "Authentication failed after reconnecting. Please reload the page to reauthenticate"

Reply:
Ok. Make it actionable. Add a refresh page button then.